src/Framework/Microsoft.Build.Framework.csproj

```diff
@@ -39,6 +39,9 @@
     <Compile Include="..\Shared\BinaryWriterExtensions.cs">
       <Link>Shared\BinaryWriterExtensions.cs</Link>
     </Compile>
+    <Compile Include="..\Shared\IMSBuildElementLocation.cs">
+      <Link>Shared\IMSBuildElementLocation.cs</Link>
```

Comment:
I would suggest to not use `..\Shared` directory and place the file directly to `Framework` project directory. It would be nice to incrementally remove the `Shared` directory completely in the future.

Reply:
Unfortunately, it's not easy to move it to Framework due to dependencies
https://github.com/dotnet/msbuild/issues/10544